### PR TITLE
Google oauth added

### PR DESCRIPTION
**Pull request by VL4DYSL4V**

### CODE REVIEW

File: slidez-core/build.gradle [hunk 1]
```diff
@@ -32,8 +32,15 @@ dependencies {
 
 	implementation 'org.springframework.boot:spring-boot-starter-actuator'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
-	implementation 'org.springframework.boot:spring-boot-starter-websocket'
+    implementation 'org.springframework.boot:spring-boot-starter-websocket'
 	implementation 'org.springframework.boot:spring-boot-starter-security'
+    implementation 'org.springframework.boot:spring-boot-starter-oauth2-client'
+	implementation group: 'org.springdoc', name: 'springdoc-openapi-ui', version: '1.5.9'
+	implementation group: 'org.springdoc', name: 'springdoc-openapi-data-rest', version: '1.5.9'
+	implementation group: 'org.springdoc', name: 'springdoc-openapi-security', version: '1.5.9'
```
Comment (VolodymyrYavorskyi) [inline slidez-core/build.gradle:40]: duplicate dependencies

**Review by VolodymyrYavorskyi — APPROVED** [review]

**Review by VolodymyrYavorskyi — APPROVED** [review]

**Comment by VL4DYSL4V** [comment]
In staging properties you have to add "oauth2.redirect-uri" and "oauth2.client-id"

File: slidez-fe-core/src/services/auth/auth-service.ts [hunk 2]
```diff
@@ -4,11 +4,18 @@ import { RegisterDto } from '../../containers/user/dto/RegisterDto'
...
+import { TokenDto } from '../../containers/user/dto/TokenDto'
 
 const JWT = 'jwt'
+const getAuthUrl = (endpoint: string) => {
+    return `http://localhost:5000/auth/${endpoint}`
+}
+const loginWithOAuthGoogleUrl: string =
+    'http://localhost:5000/auth/login/google'
+const registerWithOAuthGoogleUrl: string =
+    'http://localhost:5000/auth/register/google'
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/services/auth/auth-service.ts]: Don't use http://localhost:5000 in code, on staging this route will be different and it won't work

File: slidez-fe-core/src/services/auth/google-oauth.enum.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const GoogleOAuth = {
+    GOOGLE_CLIENT_ID:
+        '873588565253-jiec1ha3ghuf593g2kipg7aqjoaes2ci.apps.googleusercontent.com',
+    GOOGLE_COOKIE_POLICY: 'single_host_origin',
+    GOOGLE_REDIRECT_URI: 'http://localhost:3000/',
+}
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/services/auth/google-oauth.enum.ts]: Move it to env variables(Id + URI)
And it's not enum

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**